cmd/xl-sets.go

```diff
@@ -421,52 +428,40 @@ func (s *xlSets) StorageInfo(ctx context.Context, local bool) StorageInfo {
 		return storageInfo
 	}
 
-	storageDisks, dErrs := initStorageDisksWithErrors(s.endpoints)
-	defer closeStorageDisks(storageDisks)
-
-	formats, sErrs := loadFormatXLAll(storageDisks)
-
-	combineStorageErrors := func(diskErrs []error, storageErrs []error) []error {
-		for index, err := range diskErrs {
-			if err != nil {
-				storageErrs[index] = err
-			}
-		}
-		return storageErrs
-	}
-
-	errs := combineStorageErrors(dErrs, sErrs)
-	drivesInfo := formatsToDrivesInfo(s.endpoints, formats, errs)
+	s.xlDisksMu.RLock()
```

Comment:
you can do RLock/RUnlock inside Copy()

Reply:
Copy is not an xlSets method, so not possible.